src/Serializer/AbstractSerializer.php

```diff
@@ -261,17 +261,44 @@ protected function serializeValue($value)
     }
 
     /**
+     * This method is provided as a non-BC upgrade of serializeCallable,
+     * since using the callable type raises a deprecation in some cases.
+     *
+     * @param callable $callable
+     *
+     * @return string
+     */
+    protected function serializeCallableWithoutTypeHint($callable): string
+    {
+        if (!\is_callable($callable)) {
+            throw new \InvalidArgumentException('Expecting callable, got ' . \gettype($callable));
+        }
+
+        if (\is_string($callable) && !\function_exists($callable)) {
+            return $callable;
+        }
+
+        return $this->serializeCallable($callable);
+    }
+
+    /**
+     * @internal Use serializeCallableWithoutTypeHint instead (no type in argument)
+     *
      * @param callable $callable
      *
      * @return string
      */
-    protected function serializeCallable(callable $callable): string
+    protected function serializeCallable(/* callable type to be removed in 3.0, see #821 => */callable $callable): string
     {
+        if (!\is_callable($callable)) {
+            throw new \InvalidArgumentException('Expecting callable, got ' . \gettype($callable));
```

Comment:
Please use `sprintf` and handle the fact that `$callable` may be either an object or a scalar